### PR TITLE
175168098 update Colyseus and remove MongoDB dependency

### PR DESCRIPTION
**Pull request by dstrawberrygirl**
This includes new version of the network stack. When reviewing, you can filter to only show the changes in the last four commits to improve readability

### CODE REVIEW

File: Assets/Scripts/Network/ColyseusClient.cs [hunk 1]
```diff
@@ -93,142 +91,123 @@ public async void ConnectToServer(string serverEndpoint, string username, string
...
         if (IsConnected)
         {
-            await LeaveRoom();
+            LeaveRoom();
             if (players != null)    
             {
                 players.Clear();
             }
-            client.Auth.Logout();
+            // client.Auth.Logout();
```
Comment (knowuh) [inline Assets/Scripts/Network/ColyseusClient.cs:106]: I usually recommend removing commented out sections like this, but I guess knowing where authentication actions used to happen will help us if/when we implement some other auth strategy.

File: Assets/Scripts/Network/ColyseusClient.cs [hunk 2]
```diff
@@ -93,142 +91,123 @@ public async void ConnectToServer(string serverEndpoint, string username, string
...
-        else
+    }
+    async void LeaveRoom()
+    {
+        CCDebug.Log("closing connection");
+        await room.Leave(true);
+        room = null;
+    }
+
+    public NetworkPlayer GetPlayerById(string username)
```
Comment (knowuh) [inline Assets/Scripts/Network/ColyseusClient.cs:192]: I remember being puzzled by this in the past: `GetPlayerById()` and `getPlayer()` above seem really similar.  I almost wonder if this method should be renamed getPlayerByUsername?

**Review by knowuh — APPROVED** [review]
Looks good to me.  👍

File: Assets/Scripts/Network/ColyseusClient.cs (hunk 2 above)
Comment (dstrawberrygirl) [reply Assets/Scripts/Network/ColyseusClient.cs:192]: Well spotted! I had to switch things up a bit in the update, so the key in the players dictionary is now the playerId, and it turned out we had two methods doing essentially the same thing (one more efficiently than the other!). Have now combined into one method called GetPlayerById.

**Pull request merged**